pubsub: panic when using tracing due to nil Span interface

## Client

PubSub v2.4.0

## Environment

Go v1.25.8

## Code and Dependencies

```go
// Creating a client with otel tracing enabled:
client, err := pubsub.NewClientWithConfig(
	ctx,
	projectID,
	&pubsub.ClientConfig{
		EnableOpenTelemetryTracing: true,
	},
)

// Pulling messages...
```

</details>

## Expected behavior

No panics in the pubsub library.

## Actual behavior

Nil interface panic when the v2 subscriber calls [`ccSpan.End()`](https://github.com/googleapis/google-cloud-go/blob/25079346ab69a3f5dedce34b3a9ab48b04686632/pubsub/v2/subscriber.go#L427).
This started happening after the client had been running for a while.

```text
panic: runtime error: invalid memory address or nil pointer dereference                                                                               
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0xdfd902]                                                                                                                                                                                       
                                                                                                                                                                                                                                                              
goroutine 408 [running]:                                                                                                                                                                                                                                      
cloud.google.com/go/pubsub/v2.(*Subscriber).Receive.func2()                                                                                                                                                                                                   
    cloud.google.com/go/pubsub/v2@v2.4.0/subscriber.go:397 +0x682                                                                                                                                                                                             
golang.org/x/sync/errgroup.(*Group).Go.func1()                                                                                                                                                                                                                
    golang.org/x/sync@v0.19.0/errgroup/errgroup.go:93 +0x50                                                                                                                                                                                                   
created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 312                                                                                                                                                                                            
    golang.org/x/sync@v0.19.0/errgroup/errgroup.go:78 +0x95
```

This happens because `ccSpan` is never assigned when `iter.activeSpans.Load(ackh.ackID)` returns false.
https://github.com/googleapis/google-cloud-go/blob/25079346ab69a3f5dedce34b3a9ab48b04686632/pubsub/v2/subscriber.go#L406-L428

--- a/pubsub/v2/subscriber.go
+++ b/pubsub/v2/subscriber.go
@@ -423,7 +423,7 @@ func (s *Subscriber) Receive(ctx context.Context, f func(context.Context, *Messa
 						// Return nil if the context is done, not err.
 						return nil
 					}
-					if iter.enableTracing {
+					if iter.enableTracing && ccSpan != nil {
 						ccSpan.End()
 					}
 

